reflect_struct_decoder.go

```diff
@@ -51,7 +52,7 @@ func createStructDecoder(ctx *ctx, typ reflect2.Type, fields map[string]*structF
 		return &skipObjectDecoder{typ}
 	case 1:
 		for fieldName, fieldDecoder := range fields {
-			fieldHash := calcHash(fieldName)
+			fieldHash := calcHash(fieldName, ctx.caseSensitive())
```

Comment:
Does this code above also make decoding case-insensitive?
``` 
for k, binding := range bindings {
  if _, found := fields[strings.ToLower(k)]; !found {
    fields[strings.ToLower(k)] = binding.Decoder.(*structFieldDecoder)
  }
}
```

Reply:
A test case that exercises positive and negative cases would be good

Reply:
Opened https://github.com/json-iterator/go/pull/285

Reply:
> Does this code above also make decoding case-insensitive?

Yes, this was the culprit.